src/App.js

```diff
@@ -21,45 +23,106 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
+  // console.log(squares);
   return squares;
 };
 
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(playerOne);
+  const [isGameOver, setGameStatus] = useState(false);
+  const [winner, setWinner] = useState(winnerInitial);
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const updateOnClick = (updatedSquare) => {
+    const squaresData = squares.map((row) => {
+      row.map((square) => {
+        if (square.id === updatedSquare.id) {
+          if (square.value === '') {
+            square.value = player;
```

Comment:
Notice that this update modifies the same square data as is currently stored in the board state. And even though a new squareData is calculated, it is never used to update the squares. This code winds up cross-contaminating the board state rather than updating it. The thing that ends up causing React to re-render is the setWinner/setGameStatus, or setPlayer calls. These also invalidate the current state so that the entire control gets redrawn, so we end up seeing changes in the board state, even though React is never told there is a change there.

We should be vigilant about these sorts of leaky changes so that we don't modify the current state behind React's back.